lib/sync/pull-request.js

```diff
@@ -1,57 +1,91 @@
+const url = require('url');
 const transformPullRequest = require('./transforms/pull-request');
-const { getPullRequests: getPullRequestQuery } = require('./queries');
-const { logger } = require('probot/lib/logger');
-
 const statsd = require('../config/statsd');
 
+/**
+ * @typedef {object} RepositoryObject
+ * @property {number} id
+ * @property {string} name
+ * @property {object} owner
+ * @property {string} owner.login
+ * @property {string} html_url
+ * @property {string} full_name
+ */
+
+/**
+ * Find the next page number from the response headers.
+ *
+ * @param {{link: string}} headers - The headers from the API response.
+ * @returns {number|undefined} The next page number.
+ */
+function getNextPage(headers = {}) {
```

Comment:
Does probot's octokit not autopaginate?

Reply:
It does! But we're chunking the work ourselves, every 20 items we save our state and queue up another job. We do this to avoid having a job that runs for 20 hours and the fails.

Reply:
Do we pick up where it left off?

Reply:
We do! That's what we're using the page number (in the cursor) for.